lib/room.cpp

```diff
@@ -1293,6 +1313,17 @@ const StateEventBase* Room::getCurrentState(const QString& evtType,
     return d->getCurrentState({ evtType, stateKey });
 }
 
+const QList<const StateEventBase*>*
+Room::stateEventsOfType(const QString& evtType) const
+{
+    return d->stateEventsOfType(evtType);
+}
+
+const QHash<StateEventKey, const StateEventBase*> Room::stateEvents() const
```

Comment:
`QHash` is implicily-shared so copying it is cheap - but it could just be a reference, couldn't it? (Assuming that `Q_INVOKABLE` doesn't work, see the next comment.)
Also: `stateEvents()` is not specific enough, since `Room` actually maintains several (two, as of now) sets of state events. The base state is not exposed in the API as yet but it might be in the future - e.g. as a part of the protected API.
```suggestion
const QHash<StateEventKey, const StateEventBase*>& Room::currentState() const
```